pennylane/templates/state_preparations/mottonen.py

```diff
@@ -87,36 +88,47 @@ def _compute_theta(alpha):
     k = np.log2(alpha.shape[0])
     factor = 2 ** (-k)
 
-    theta = sparse.dok_matrix(alpha.shape, dtype=np.float64)  # type: sparse.dok_matrix
+    theta = np.zeros(alpha.shape, dtype=np.float64)
 
     for row in range(alpha.shape[0]):
         # Use transpose of M:
-        entry = sum([_matrix_M_entry(col, row) * a for (col, _), a in alpha.items()])
+        entry = sum(_matrix_M_entry(col, row) * alpha[col] for col in range(len(alpha)))
         entry *= factor
         if abs(entry) > 1e-6:
-            theta[row, 0] = entry
+            theta[row] = entry
 
     return theta
 
 
 def _uniform_rotation_dagger(gate, alpha, control_wires, target_wire):
-    """Applies a given inverse rotation to the target qubit
-    that is uniformly controlled by the control qubits.
+    """Applies a uniformly-controlled rotation to the target qubit.
+
+    A uniformly-controlled rotation is a sequence of multi-controlled
+    rotations, each of which is conditioned on the control qubits being in a different state.
+    For example, a uniformly-controlled rotation with two control qubits describes a sequence of
+    four multi-controlled rotations, each applying the rotation only if the control qubits
+    are in states :math:`|00\rangle`, :math:`|01\rangle`, :math:`|10\rangle`, and :math:`|11\rangle`, respectively.
+
+    To implement a uniformly-controlled rotation using single qubit rotations and CNOT gates,
+    a decomposition based on gray codes is used. For this purpose, the multi-controlled rotation
+    angles alpha have to be converted into a set of non-controlled rotation angles theta.
+
+    For more details, see `Möttönen and Vartiainen (2005), Fig 7a<https://arxiv.org/pdf/quant-ph/0504100.pdf>`_.
 
     Args:
-        gate (~.Operation): gate to be applied, needs to have exactly
-            one parameter
-        alpha (array[float]): alpha parameters
+        gate (~.Operation): gate to be applied, needs to have exactly one parameter
+        alpha (array[float]): angles to decompose the uniformly-controlled rotation into multi-controlled rotations
```

Comment:
I find this description hard to parse; alpha are like the initial rotation angles before the decomposition?

Reply:
Yes, in some sense the alpha are rotation angles of the decomposition/representation of the uniformly controlled rotations into multi-controlled rotations (this is kind of their definition), and the theta are angles when the uniformly controlled rotations are decomposed into something else. It's hard to express well without pics.